test/integration/play/test_play.py

```diff
@@ -37,100 +29,85 @@ def test_play_non_existent_uri(tt_cmd, tmp_path):
     assert re.search(r"no connection to the host", output)
 
 
-def test_play_non_existent_file(tt_cmd, tmp_path, test_instance):
-    # Run play with non-existent file.
-    cmd = [tt_cmd, "play", "127.0.0.1:" + test_instance.port, "path-to-non-existent-file"]
-    rc, output = run_command_and_get_output(cmd, cwd=tmp_path)
-    assert rc == 1
-    assert re.search(r"No such file or directory", output)
-
-
-def test_play_test_remote_instance(tt_cmd, test_instance):
+@pytest.mark.parametrize("args, play_error", [
+    (
+        # Testing with unset uri and .xlog or .snap file.
+        "",
+        "required to specify an URI and at least one .xlog or .snap file",
+    ),
+    (
+        "path-to-non-existent-file",
+        "No such file or directory",
+    ),
+    (
+        ("test.xlog", "--timestamp=abcdef", "--space=999"),
+        'failed to parse a timestamp: parsing time "abcdef"',
+    ),
+    (
+        ("test.xlog", "--timestamp=2024-11-14T14:02:36.abc", "--space=999"),
+        'failed to parse a timestamp: parsing time "2024-11-14T14:02:36.abc"',
+    ),
+])
+def test_play_test_remote_instance_timestamp_failed(tt_cmd, test_instance, args, play_error):
     # Play .xlog file to the remote instance.
-    cmd = [tt_cmd, "play", "127.0.0.1:" + test_instance.port, "test.xlog", "--space=999"]
-    rc, output = run_command_and_get_output(cmd, cwd=test_instance._tmpdir)
-    assert rc == 0
-    assert re.search(r"Play result: completed successfully", output)
-
-    # Testing played .xlog file from the remote instance.
-    cmd = [tt_cmd, "cat", "00000000000000000000.xlog", "--space=999"]
-    rc, output = run_command_and_get_output(cmd, cwd=test_instance._tmpdir)
-    assert rc == 0
-    assert re.search(r"space_id: 999", output)
-    assert re.search(r"[1, 'Roxette', 1986]", output)
-    assert re.search(r"[2, 'Scorpions', 2015]", output)
-    assert re.search(r"[3, 'Ace of Base', 1993]", output)
-
-
-TEST_PLAY_TIMESTAMP_PARAMS_CCONFIG = ("input, play_result, found, not_found")
-
-
-def make_test_play_timestamp_param(
-    input="",
-    play_result=0,
-    found={},
-    not_found={},
-):
-    return pytest.param(input, play_result, found, not_found)
+    cmd = [tt_cmd, "play", f"127.0.0.1:{test_instance.port}"]
+    cmd.extend(args)
+    rc, play_output = run_command_and_get_output(cmd, cwd=test_instance._tmpdir)
+    assert rc == 1
+    assert play_error in play_output
 
 
-@pytest.mark.parametrize(TEST_PLAY_TIMESTAMP_PARAMS_CCONFIG, [
-    make_test_play_timestamp_param(
-        input="abcdef",
-        play_result=1,
-        found={"failed to parse a timestamp: parsing time \"abcdef\""},
+@pytest.mark.parametrize("args, found, not_found", [
```

Comment:
What `not_found` does here? It's a sign of wrong test, again the same.

Reply:
After `tt play` command wal-file contains records with updated `timestamp`, so we could check the existence or the absence of records only by their data.

Reply:
Rewrite to directly get data from tarantool instance.